lib/ui/fake_flutter/diagnosticable.dart

```diff
@@ -0,0 +1,2941 @@
+      'description': toDescription(),
+      'type': runtimeType.toString(),
+    };
+    if (name != null) data['name'] = name;
+
+    if (!showSeparator) data['showSeparator'] = showSeparator;
+    if (level != DiagnosticLevel.info) data['level'] = describeEnum(level);
+    if (showName == false) data['showName'] = showName;
+    if (emptyBodyDescription != null)
+      data['emptyBodyDescription'] = emptyBodyDescription;
+    if (style != DiagnosticsTreeStyle.sparse)
+      data['style'] = describeEnum(style);
+
+    bool hasChildren = getChildren().isNotEmpty;
+    if (hasChildren) data['hasChildren'] = hasChildren;
+
+    if (linePrefix?.isNotEmpty == true) data['linePrefix'] = linePrefix;
+    return data;
+  }
+
+  /// Returns a string representation of this diagnostic that is compatible with
+  /// the style of the parent if the node is not the root.
+  ///
+  /// `parentConfiguration` specifies how the parent is rendered as text art.
+  /// For example, if the parent places all properties on one line, the
+  /// [toString] for each property should avoid line breaks if possible.
+  ///
+  /// `minLevel` specifies the minimum [DiagnosticLevel] for properties included
+  /// in the output.
+  @override
+  String toString({
+    TextTreeConfiguration parentConfiguration,
+    DiagnosticLevel minLevel = DiagnosticLevel.info,
+  }) {
+    assert(style != null);
+    assert(minLevel != null);
+    if (style == DiagnosticsTreeStyle.singleLine)
+      return toStringDeep(
+          parentConfiguration: parentConfiguration, minLevel: minLevel);
+
+    final String description =
+        toDescription(parentConfiguration: parentConfiguration);
+
+    if (name == null || name.isEmpty || !showName) return description;
+
+    return description.contains('\n')
+        ? '$name$_separator\n$description'
+        : '$name$_separator $description';
+  }
+
+  /// Returns a configuration specifying how this object should be rendered
+  /// as text art.
+  @protected
+  TextTreeConfiguration get textTreeConfiguration {
+    assert(style != null);
+    switch (style) {
+      case DiagnosticsTreeStyle.dense:
+        return denseTextConfiguration;
+      case DiagnosticsTreeStyle.sparse:
+        return sparseTextConfiguration;
+      case DiagnosticsTreeStyle.offstage:
+        return dashedTextConfiguration;
+      case DiagnosticsTreeStyle.whitespace:
+        return whitespaceTextConfiguration;
+      case DiagnosticsTreeStyle.transition:
+        return transitionTextConfiguration;
+      case DiagnosticsTreeStyle.singleLine:
+        return singleLineTextConfiguration;
+      case DiagnosticsTreeStyle.indentedSingleLine:
+        return singleLineTextConfigurationIndented;
+      case DiagnosticsTreeStyle.shallow:
+        return shallowTextConfiguration;
+      case DiagnosticsTreeStyle.error:
+        return errorTextConfiguration;
+      case DiagnosticsTreeStyle.truncateChildren:
+        return null;
+    }
+    return null;
```

Comment:
Could put this in a default case.

Reply:
this code isn't for review here. it is just forked from package flutter so ignoring your valid comments :)